Isfulladmin py

will-moore opened the pull request:
# What this PR does

Fixes test from #5483 (replaces that PR).

# Testing this PR

1. Check test is green

cc @dominikl

dominikl committed:
Added isFullAdmin method to gateway

dominikl committed:
Added isFullAdmin integration test

will-moore committed:
Simplify conn.isFullAdmin()

will-moore committed:
Fix test_full_admin_privileges() creating new users

mtbc commented on components/tools/OmeroPy/src/omero/gateway/__init__.py:
```diff
@@ -2324,6 +2324,19 @@ def isAdmin(self):
+        """
+
+        if self.getEventContext().isAdmin:
+            allPrivs = list(self.getEnumerationEntries('AdminPrivilege'))
```
Would `allPrivs` be worth caching, or noting upon connection? (Oh, or does `getEnumerationEntries` already do that?)

will-moore replied:
I guess it depends how expensive it is to ```getEnumerationEntries()```. I don't imagine you'd be calling this multiple times in normal usage?

mtbc replied:
Fair enough, not much more expensive than the server roundtrip overhead. Not a blocker.

dominikl commented:
👍 Thanks @will-moore !

joshmoore committed:
admin_only: default to full_admin semantics

The semantics of the admin_only have been changed by the
introduction of restricted admins. Whereas previously,
isAdmin (i.e. membership in the system group) implied
complete power, restricted admins are *also* in the
system group but may be missing a non-future-proof list
of privileges.

This restores the previous behavior of having admin_only
require full permissions by acquiring the current list
of privileges and checking against those of the user.

If the new "minimial-restricted-admin" logic is needed,
then use `@admin_only(full_admin=False)`.

See: https://trello.com/c/4dvUYSjS/126-definition-of-ultimate-admin

joshmoore committed:
Review all `admin_only()` usages

This opens up the access to some of the admin_only decorators
by passing `full_admin=False`.

will-moore committed:
Merge pull request #58 from joshmoore/isfulladmin_py

admin_only: default to full_admin semantics

joshmoore commented:
Closing and re-opening now that https://github.com/openmicroscopy/openmicroscopy/pull/5498 is merged

pwalczysko commented:
See https://github.com/will-moore/openmicroscopy/pull/58#issuecomment-328887423 - possible broken tests here.

will-moore commented:
https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-integration-Python27/656/testReport/OmeroPy.test.integration.clitest.test_ldap/TestLDAP/testAdminOnly_list_/
failing...

will-moore committed:
bin/omero ldap list  is not admin-only. Update test

pwalczysko commented:
The commit https://github.com/openmicroscopy/openmicroscopy/pull/5494/commits/504625cf42015d626b03733b419a6a4413ee4c0e had the desired effect, the tests are passing now. https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-integration-python/lastCompletedBuild/testReport/OmeroPy.test.integration.clitest.test_ldap/TestLDAP/ and
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-Python27/lastCompletedBuild/testReport/OmeroPy.test.integration.clitest.test_ldap/TestLDAP/

Happy with the change as well, although the test was fully removed for the ``list`` subcommand, it was never doing anything else but checking that it is ``AdminOnly`` in the first place, so no empoverishment of the test suite happened.

will-moore commented:
@joshmoore was going to have a final check of this...?

joshmoore commented:
https://github.com/openmicroscopy/openmicroscopy/blob/v5.3.4/components/server/src/ome/logic/AdminImpl.java#L330 (backed by https://github.com/openmicroscopy/openmicroscopy/blob/v5.3.4/components/server/src/ome/logic/LdapImpl.java#L657) does allow regular users to list others' distinguised name, but this seems fine to me. :+1:
